cmd/dbconv/dbconv/dbconv.go

```diff
@@ -0,0 +1,151 @@
+package dbconv
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type DBConverter struct {
+	config *DBConvConfig
+	stats  Stats
+}
+
+func NewDBConverter(config *DBConvConfig) *DBConverter {
+	return &DBConverter{
+		config: config,
+	}
+}
+
+func openDB(config *DBConfig, name string, readonly bool) (ethdb.Database, error) {
+	return rawdb.Open(rawdb.OpenOptions{
+		Type:      config.DBEngine,
+		Directory: config.Data,
+		// we don't open freezer, it doesn't need to be converted as it has format independent of db-engine
+		// note: user needs to handle copying/moving the ancient directory
+		AncientsDirectory:  "",
+		Namespace:          config.Namespace,
+		Cache:              config.Cache,
+		Handles:            config.Handles,
+		ReadOnly:           readonly,
+		PebbleExtraOptions: config.Pebble.ExtraOptions(name),
+	})
+}
+
+func (c *DBConverter) Convert(ctx context.Context) error {
+	var err error
+	src, err := openDB(&c.config.Src, "src", true)
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+	dst, err := openDB(&c.config.Dst, "dst", false)
+	if err != nil {
+		return err
+	}
+	defer dst.Close()
+	c.stats.Reset()
+	log.Info("Converting database", "src", c.config.Src.Data, "dst", c.config.Dst.Data, "db-engine", c.config.Dst.DBEngine)
+	it := src.NewIterator(nil, nil)
+	defer it.Release()
+	batch := dst.NewBatch()
+	entriesInBatch := 0
+	for it.Next() && ctx.Err() == nil {
+		if err = batch.Put(it.Key(), it.Value()); err != nil {
+			return err
+		}
+		entriesInBatch++
+		if batchSize := batch.ValueSize(); batchSize >= c.config.IdealBatchSize {
+			if err = batch.Write(); err != nil {
+				return err
+			}
+			c.stats.AddEntries(int64(entriesInBatch))
+			c.stats.AddBytes(int64(batchSize))
```

Comment:
I misread stats as status and thought the function names confused me. Maybe change names to something like LogEntried/LogBytes so it will be clear they deal with stats only and don't actualy manage entries/bytes?